rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -120,10 +120,11 @@ def __init__(self, topic, client_id, callback, node_handle, msg_type=None, raw=F
         )
 
         infos = node_handle.get_publishers_info_by_topic(topic)
-        if any(pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos):
+
+        if all(pub.qos_profile.durability == DurabilityPolicy.TRANSIENT_LOCAL for pub in infos):
             qos.durability = DurabilityPolicy.TRANSIENT_LOCAL
             qos.reliability = ReliabilityPolicy.RELIABLE
-        if any(pub.qos_profile.reliability == ReliabilityPolicy.BEST_EFFORT for pub in infos):
+        if all(pub.qos_profile.reliability == ReliabilityPolicy.BEST_EFFORT for pub in infos):
```

Comment:
This one should still be `any`. We want to force the best effort reliability if **any** of the publishers use best effort qos for best compatibility.
```suggestion
        if any(pub.qos_profile.reliability == ReliabilityPolicy.BEST_EFFORT for pub in infos):
```